src/Service/BroadcastsService.php

```diff
@@ -84,4 +87,41 @@ function () use ($serviceId, $startDate, $endDate, $limit, $page) {
             }
         );
     }
+
+    /**
+     * @return Broadcast[]
+     */
+    public function findUpcomingByService(
+        Service $service,
+        ?int $limit = self::DEFAULT_LIMIT,
+        int $page = self::DEFAULT_PAGE,
+        $ttl = CacheInterface::NORMAL
+    ): array {
+        $key = $this->cache->keyHelper(
+            __CLASS__,
+            __FUNCTION__,
+            $service->getDbId(),
+            $limit,
+            $page,
+            $ttl
+        );
+
+        // TODO set the cache time based on the end date of the first returned item, so that the cache expires once the
```

Comment:
Is there a ticket for this? Or should it be done as part of this PR?

Reply:
Should be done as part of this PR please :)